third-culture-software/bhima
# Improve shipment tracking logging

Improve the shipment tracking logging
- Automatically adds a shipment tracking node when you create, edit, mark ready to ship, do stock exit, do stock entry, and complete a shipment.
- Enable the user to add shipment tracking notes at any time
- Adds option to mark a partially delivered shipment as complete (with appropriate tracking log note).  This is for cases where some items were lost in transit.
- When stock exits or entries are invoked from the shipping registry, we automatically return to the registry when the action is complete.

Closes https://github.com/IMA-WorldHealth/bhima/issues/6582
Closes https://github.com/IMA-WorldHealth/bhima/issues/6581

**TESTING**
- Create a shipment and do each of these steps. After each step, check the "Tracking log" from the action menu to observe the notes
   - Create
   - Edit (update packing list)
   - Mark ready to ship
   - Stock Exit
   - Stock Entry - PARTIAL
   - Stock Entry - for the rest of the items.  This will mark the shipment as complete and add an entry in the tracking log
- Repeat the process above, but after doing a partial entry, mark the shipment as complete and check the tracking log
